src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.cpp

```diff
@@ -215,6 +217,32 @@ void GenericThreadStackManagerImpl_OpenThread<ImplClass>::_OnPlatformEvent(const
 
 #endif // CHIP_DETAIL_LOGGING
 
+#if CHIP_SYSTEM_CONFIG_USE_OPEN_THREAD_ENDPOINT
+        if (event->ThreadStateChange.RoleChanged || event->ThreadStateChange.AddressChanged)
+        {
+            bool isInterfaceUp;
+            isInterfaceUp = GenericThreadStackManagerImpl_OpenThread<ImplClass>::IsThreadInterfaceUpNoLock();
+            // Post an event signaling the change in Thread interface connectivity state.
+            {
+                ChipDeviceEvent event;
+                event.Clear();
+                event.Type                            = DeviceEventType::kThreadConnectivityChange;
+                event.ThreadConnectivityChange.Result = (isInterfaceUp) ? kConnectivity_Established : kConnectivity_Lost;
+                CHIP_ERROR status                     = PlatformMgr().PostEvent(&event);
+                if (status != CHIP_NO_ERROR)
+                {
+                    ChipLogError(DeviceLayer, "Failed to post Thread connectivity change: %" CHIP_ERROR_FORMAT, status.Format());
+                }
+            }
+
+            // Refresh Multicast listening
+            if (GenericThreadStackManagerImpl_OpenThread<ImplClass>::IsThreadAttachedNoLock())
+            {
+                ChipLogDetail(DeviceLayer, "Thread Attached updating Multicast address");
+                Server::GetInstance().RejoinExistingMulticastGroups();
```

Comment:
Actually, why this cannot be done in `Server::Init` like for other platforms?

Reply:
@Damian-Nordic can you give more insight information since this 
CHIP_SYSTEM_CONFIG_USE_OPEN_THREAD_ENDPOINT is Generic !

Reply:
Take a look at `Server::Init` which calls `RejoinExistingMulticastGroups` for all platforms but those that define `CHIP_SYSTEM_CONFIG_USE_OPEN_THREAD_ENDPOINT`. I wonder where this discrepancy comes from.

Reply:
So in one case you do RejoinExistingMulticastGroups :

-  !CHIP_SYSTEM_CONFIG_USE_OPEN_THREAD_ENDPOINT at Server:Init once 
-  CHIP_SYSTEM_CONFIG_USE_OPEN_THREAD_ENDPOINT  each time at _OnPlatformEvent with certain events if notLocked

this sound indeed like a something that will require a more universal fix...

Reply:
I m not qualify with Thread ... but i think an issue for a later TODO should be created

Reply:
This need to be rerun for multicast subscription at each OpenThread network update, otherwise should the OTBR change/update/etc. the device won't received group messages. This could also be done outside of the `CHIP_SYSTEM_CONFIG_USE_OPEN_THREAD_ENDPOINT ` define in this case. 

Also, it shouldn't be done for OpenThread in the Server::Init since most of the time, the thread interface isn't up and running or fully initialized when the method is called. At least that's what we observed on our platform as we didn't get any feedback from other platform manufacturer on this subject.